google/cloud/internal/curl_impl.cc

```diff
@@ -654,6 +654,7 @@ Status CurlImpl::MakeRequest(CurlImpl::HttpMethod method,
   using HttpMethod = CurlImpl::HttpMethod;
   handle_.SetOption(CURLOPT_CUSTOMREQUEST, HttpMethodAsChar(method));
   handle_.SetOption(CURLOPT_UPLOAD, 0L);
+  handle_.SetOption(CURLOPT_FOLLOWLOCATION, 1L);
```

Comment:
Do we want this for *all* requests?  Should this be an option, maybe in the `*_internal::` namespace?